app/graphql/types/weather_type.rb

```diff
@@ -9,114 +9,99 @@ class Types::WeatherObject < Types::BaseObject
   field_class Types::WeatherField
 end
 
-class Types::WeatherDailyDataType < Types::WeatherObject
-  field :time, Types::UnixDateTimeType
-  field :summary, String
-  field :icon, String
-  field :sunriseTime, Types::UnixDateTimeType
-  field :sunsetTime, Types::UnixDateTimeType
-  field :precipIntensity, Float
-  field :precipIntensityMax, Float
-  field :precipIntensityMaxTime, Types::UnixDateTimeType
-  field :precipProbability, Float
-  field :precipType, String
-  field :temperatureHigh, Float
-  field :temperatureHighTime, Types::UnixDateTimeType
-  field :temperatureLow, Float
-  field :apparentTemperatureHigh, Float
-  field :apparentTemperatureLow, Float
-  field :dewPoint, Int
-  field :humidity, Float
-  field :pressure, Float
-  field :windSpeed, Float
-  field :windGust, Float
-  field :windBearing, Int
-  field :cloudCover, Float
-  field :uvIndex, Int
-  field :ozone, Float
-  field :temperatureMin, Float
-  field :temperatureMax, Float
-  field :apparentTemperatureMin, Float
-  field :apparentTemperatureMax, Float
+class Types::WeatherTemperatureDataType < Types::WeatherObject
+  field :day, Types::KelvinToFarenheitType
+  field :night, Types::KelvinToFarenheitType
+  field :morn, Types::KelvinToFarenheitType
+  field :eve, Types::KelvinToFarenheitType
+  field :min, Types::KelvinToFarenheitType
+  field :max, Types::KelvinToFarenheitType
 end
 
-class Types::WeatherHourlyDataType < Types::WeatherObject
-  field :time, Types::UnixDateTimeType
-  field :summary, String
-  field :icon, String
-  field :precipIntensity, Float
-  field :precipProbability, Float
-  field :precipType, String
-  field :temperature, Float
-  field :apparentTemperature, Float
-  field :dewPoint, Float
-  field :humidity, Float
-  field :pressure, Float
-  field :windSpeed, Float
-  field :windGust, Float
-  field :windBearing, Int
-  field :cloudCover, Float
-  field :uvIndex, Int
-  field :visibility, Int
-  field :ozone, Float
+class Types::WeatherApparentTemperatureDataType < Types::WeatherObject
+  field :day, Types::KelvinToFarenheitType
+  field :night, Types::KelvinToFarenheitType
+  field :morn, Types::KelvinToFarenheitType
+  field :eve, Types::KelvinToFarenheitType
 end
 
-class Types::WeatherMinutelyDataType < Types::WeatherObject
-  field "time", Types::UnixDateTimeType
-  field "precipIntensity", Float
-  field "precipIntensityError", Float
-  field "precipProbability", Float
-  field "precipType", String
+class Types::WeatherInfoDataType < Types::WeatherObject
+  def initialize(value, context, **kwargs, &block)
+    super(value[0], context, **kwargs, &block)
```

Comment:
Maybe add comment mentioned we converting the inbound `[{ ... }]` into `{ .... }`. Specifically mentioning that its an array of one object.